Src/Core/Common/ExpressionExtensions.cs

```diff
@@ -129,9 +131,17 @@ public static class ExpressionExtensions
                         node = null;
                         break;
 
+                    case ExpressionType.Call:
+                        var methodCallExpression = node as MethodCallExpression;
+                        if (methodCallExpression?.Method.Name != "get_Item" || methodCallExpression.Arguments.Count != 1 || !(methodCallExpression.Arguments[0] is ConstantExpression))
+                            throw new ArgumentException(unsupportedExpressionMessage);
```

Comment:
To avoid duplication, because the exception message is now used in a couple of places (here and the `default` case).

Updated to add braces.